packages/paste-website/src/pages/components/toast/index.mdx

```diff
@@ -144,38 +150,114 @@ Here are ways that you can compose neutral, warning, error, success, and dismiss
+
+- A `<Toaster />` component that handles rendering toasts into the DOM.
+- A `useToaster` hook that handles adding, removing, auto dismissing and state.
+
+#### useToaster hook
+
+The easiest way to use the Toaster is to use the `useToaster` hook. It will handle adding and removing a Toast and can be used to set automatic time-outs. The `useToaster` hook will then return the props needed to render the Toasts within the `Toaster` component.
+
+`useToaster` returns 3 things;
+
+- An array of `toasts` to be rendered in the DOM.
+- A `push` method used to add a Toast to the Toaster.
+- A `pop` method used to remove a Toast from the Toaster via it's id.
+
+Below is a form you can use to create toasts dynamically using the Toaster. The form data is used to `push` a new Toast to the Toaster.
+
+<Box marginBottom="space70">
+  <Card>
+    <ToasterExample />
+  </Card>
+</Box>
+
+##### Adding a Toast
+
+You can push a Toast to the Toaster stack based on any action or event. In this example it's based on a button click:
+
+```
+const MyComponent = () => {
+  const toaster = useToaster();
+
+  const handleClick = () => {
+    toaster.push({
+      message: 'Selection was successfully saved to your profile',
+      variant: 'success',
+    })
+  }
+
+  return(
+    <>
+      <Button variant="primary" onClick={handleClick}>
+        Add a toast
+      </Button>
+      <Toaster {...toaster} />
+    </>
+  )
+}
+```
 
-For a basic method of hiding and showing, you can use the `hidden` HTML attribute. Passing a `true` or `false` value to that attribute will show or hide the Toast respectively.
+##### Removing a Toast
 
-When showing multiple toasts, place the most recent toast at the bottom of the stack.
+You can optionally provide an `id` to a toast when you push it to the stack. The ID can then be used to programmatically `pop` the Toast from the stack:
+
+```
+toaster.push({
+  message: 'Selection was successfully saved to your profile',
+  variant: 'success',
+  id: 'custom_id',
+})
+
+toaster.pop('custom_id')
+```
+
+When providing your own custom ID, make sure they are unique.
+
+##### Automatic dismissal
+
+You can optionally provide a time-out to a toast, after which the Toast is automatically dismissed. This can be done by setting a `dismissAfter` value. The value must be a valid time-out, set in milliseconds.
+
+```
+toaster.push({
+  message: 'Selection was successfully saved to your profile',
+  variant: 'success',
+  id: 'custom_id',
+  dismissAfter: 4000
+})
+```
 
 Add an automatic time-out if all these conditions are true:
 
 - The user wouldn't need to remember or copy the information in order to continue with their current flow, or any future flow.
 - The toast doesn't have any interactive content, such as a link or button.
 
-We recommend giving users about 0.5 seconds per word before timing out a toast.
+We recommend giving users about 500 milliseconds per word before timing out a toast.
+
+#### Toaster component
```

Comment:
In what scenario would someone want to use Toaster by itself? Do we need to include this section up here, or can we move it to the API section?

Reply:
It's a very specific use-case where you are controlling state for everything in the UI from the application in a redux store, for example. It might be best in the API section though, I think you might be right about that. I don't think many people will do it.